clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/SubnetController.groovy

```diff
@@ -31,10 +32,27 @@ class SubnetController {
   List<SubnetProvider> subnetProviders
 
   @RequestMapping(method = RequestMethod.GET)
-  Set<Subnet> list() {
-    subnetProviders.collectMany {
+  Map<String, Set<Subnet>> list() {
```

Comment:
Feels like we need to preserve the original `Set<Subnet>` response and avoid a breaking change. The `Subnet` should have a `type` attribute allowing a prospective caller to differentiate by type if necessary.

Similarly, it seems like filtering by `cloudProvider` could also be accomplished by a filter query parameter? That said, I don't believe we're terribly consistent on filters vs. path params so if you have a reasonable "it's like this already" argument for `/{cloudProvider}` than I'm convinced.

Gate is deserving of a `v2` treatment with consistency in mind.